86) Write a program to check if a string has a number or not without using Built-in string methods.

In [1]:
a = input("Enter a string: ")

found = False

for i in a:
    if i >= '0' and i <= '9':
        found = True
        break

if found:
    print("The given string contain a number.")
else:
    print("The given string does not contain a number.")


Enter a string: john234
The given string contain a number.


87) Convert the string into integer, without using the built-in int() function.

In [4]:
s = input("Enter a number string: ")

num = 0

for i in s:
    digit = ord(i) - ord('0')
    num = num * 10 + digit

print(num)


Enter a number string: alex
55602


88) In a cricket match, the batting team is scoring runs with some run rate. The match is
for 20 overs. Write a program to find the total runs scored by the batting.
Input :13
Output: Total runs scored in 20 overs: 260.00

In [6]:
n = int(input("Enter a number: "))

rev = 0

while n > 0:
    digit = n % 10
    rev = rev * 10 + digit
    n = n // 10

print("Number in reverse order:", rev)


Enter a number: 320
Number in reverse order: 23



89)A run is a bus conductor. His ticket machine is printing numbers in reverse order
due to a technical glitch. As a Programmer on the bus, you are asked to help him by
creating a program to display the numbers correctly.
Input :320
Output: Number in reverse order: 23

In [5]:
rate = float(input("Enter the run rate: "))
total = rate * 20
total = int(total * 100) / 100
print("Total runs scored in 20 overs:", total)


Enter the run rate: 13
Total runs scored in 20 overs: 260.0



90)Given an odd length word which should be printed from the middle of the word.




In [8]:
s = input("Enter the word: ")

n = len(s)
mid = n // 2

result = s[mid]
print(result)

for i in range(mid + 1, n):
    result += s[i]
    print(result)

for i in range(mid - 1, -1, -1):
    result += s[i]
    print(result)


Enter the word: PROGRAM
G
GR
GRA
GRAM
GRAMO
GRAMOR
GRAMORP
